aws/resource_aws_redshift_parameter_group.go

```diff
@@ -161,6 +183,12 @@ func resourceAwsRedshiftParameterGroupUpdate(d *schema.ResourceData, meta interf
 		d.SetPartial("parameter")
 	}
 
+	if tagErr := setTagsRedshift(conn, d); tagErr != nil {
```

Comment:
There was a subtle bug introduced here as the handling of the `tags` argument would trigger an error because it was not protected by an `if d.IsNewResource()` conditional and the `arn` attribute was not set yet (which showed as a `*` resource ARN in the Redshift error):

```
--- FAIL: TestAccAWSRedshiftParameterGroup_withTags (8.02s)
    testing.go:568: Step 0 error: errors during apply:

        Error: AccessDenied: User: arn:aws:iam::--OMITTED--:user/bflad is not authorized to perform: redshift:CreateTags on resource: *
```

To fix this, opted to switch `resourceAwsRedshiftParameterGroupCreate` from returning `resourceAwsRedshiftParameterGroupUpdate` to instead return `resourceAwsRedshiftParameterGroupRead` and implement setting parameters during creation:

```go
	if v := d.Get("parameter").(*schema.Set); v.Len() > 0 {
		parameters, err := expandRedshiftParameters(v.List())

		if err != nil {
			return fmt.Errorf("error expanding parameter: %s", err)
		}

		modifyOpts := redshift.ModifyClusterParameterGroupInput{
			ParameterGroupName: aws.String(d.Id()),
			Parameters:         parameters,
		}

		if _, err := conn.ModifyClusterParameterGroup(&modifyOpts); err != nil {
			return fmt.Errorf("error adding Redshift Parameter Group (%s) parameters: %s", d.Id(), err)
		}
	}
```